[v1.5][W09-B1]Car Servicing Manager (CarviciM)

whenzei opened the pull request:
@charmaineleehc @Richardson0694 @yuhongherald

eugenepeh commented on docs/AboutUs.adoc:
```diff
@@ -3,18 +3,20 @@
-=== John Doe
-image::damithc.jpg[width="150", align="left"]
-{empty}[http://www.comp.nus.edu.sg/~damithch[homepage]] [https://github.com/damithc[github]] [<<johndoe#, portfolio>>]
+=== Ow Zhen Wei
```
Please get the team to update their info as well, tag them to this PR if u can

eugenepeh commented on docs/DeveloperGuide.adoc:
```diff
@@ -800,33 +802,139 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User presses a key to change selection.
+2.  CarServicingManager changes selection.
```
`changes selection` and ... ? Is it significant?

eugenepeh commented on docs/DeveloperGuide.adoc:
```diff
@@ -800,33 +802,139 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+*Extensions*
+
+[none]
+* 1a. CarServicingManager is not focussed.
```
spelling error

eugenepeh commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,18 +948,137 @@ Use case ends.
+.  Should be able to work without internet connection.
+.  Should respond to user command within 2 seconds.
+.  Should be usable by a novice who has never used an electronic job management app before.
+.  User interface should be intuitive enough for users who are not IT-savvy.
```
This and the one above seems similar

eugenepeh commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,18 +948,137 @@ Use case ends.
+.  Should respond to user command within 2 seconds.
+.  Should be usable by a novice who has never used an electronic job management app before.
+.  User interface should be intuitive enough for users who are not IT-savvy.
+.  View should fit the screen of any computer screen size.
```
Any computer screen size is a bit ambiguous, perhaps target a specific screen resolution or ratio e.g. 16:9?

eugenepeh commented on docs/team/owzhenwei.adoc:
```diff
@@ -0,0 +1,70 @@
+
+== Summary of contributions
+
+* *Major enhancement*: added *the ability to undo/redo previous commands*
```
remember to update accordingly; you can't claim an existing feature in the address book unless you did major overhaul to it.

eugenepeh commented on src/main/java/seedu/address/logic/commands/Command.java:
```diff
@@ -24,6 +46,16 @@ public static String getMessageForPersonListShownSummary(int displaySize) {
+     */
+    public String getCommandWord(String key) throws CommandWordException {
+        requireNonNull(model);
+        return model.getCommandWords().getCommandWord(key);
```
`getCommandWords().getCommandWord` does this sounds right to you?

eugenepeh commented on src/main/java/seedu/address/logic/commands/CommandWords.java:
```diff
@@ -0,0 +1,138 @@
+    /**
+     * Moves (@code command from (@code commands) to (@code verifiedCommands). Creates a new entry if missing.
+     */
+    private void moveVerifiedWord(String command, HashMap<String, String> verifiedCommands) {
```
not very intuitive, what do you hope to achieve with this? It can be a little too `fine-grained` if this method doesn't make sense on its own.

eugenepeh commented on src/main/java/seedu/address/logic/parser/SetCommandParser.java:
```diff
@@ -0,0 +1,29 @@
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SetCommand parse(String args) throws ParseException {
+        String[] commandWords = args.split("\\s+");
```
can probably be replace the magic value `\\s+` with a constant

eugenepeh commented on src/test/java/systemtests/AddressBookSystemTest.java:
```diff
@@ -182,6 +182,22 @@ protected void assertApplicationDisplaysExpected(String expectedCommandInput, St
+     * and the person list panel displays the persons in the model correctly.
+     */
+    protected void assertApplicationDisplaysExpectedError(String expectedCommandInput, String expectedResultMessage,
+                                                     Model expectedModel) {
```
indentation

eugenepeh commented on src/main/java/seedu/address/model/job/JobNumber.java:
```diff
@@ -0,0 +1,39 @@
+    /**
+     * Initialize the next job number of the car servicing manager
+     */
+    public static void initNextJobNumber(String args) {
```
This class should probably use a singleton design.

For this method name I think `initialize` would sound better than `initNextJobNumber` if its an one-off thing.

eugenepeh reviewed (commented):
Some feedbacks for v1.0, please tag your teammates as well

eugenepeh commented on docs/UserGuide.adoc:
```diff
@@ -202,25 +230,25 @@ Format: `redo`
 === Clearing all entries : `clear`
 
-Clears all entries from the address book. +
+Clears all entries from CarviciM. +
```
Not really specific, does all entries include employees as well?

eugenepeh commented:
![image](https://user-images.githubusercontent.com/19277206/37557704-bbf5eb4c-2a43-11e8-810b-a02c633fc074.png)
Ui looks good, but ~why is there a command box above the employee list too?~
^Edit: the Ui made me mistaken that the title is a entry field.
For the remarks, is the numbering handled by your system? So the job can have multiple remarks?

Perhaps can make everything english to makes it look more appealing to the users.

eugenepeh commented on docs/DeveloperGuide.adoc:
```diff
@@ -800,33 +824,134 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+*Extensions*
+
+[none]
+* 2a. The remarks list is empty.
```
This should be 1a. because it prevents you from going to 2.

eugenepeh commented on docs/DeveloperGuide.adoc:
```diff
@@ -800,33 +824,134 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+Use case resumes at step 3.
+
+[discrete]
+=== Use case: Tab auto-complete commands
```
I don't quite get this one, does it auto-complete the command word or does it help you fill up the suggested command usage too?

eugenepeh reviewed (commented):
Good job, the documentations seems to be well updated.

eugenepeh commented on src/main/java/seedu/address/model/job/JobList.java:
```diff
@@ -0,0 +1,62 @@
+     *
+     * @throws JobNotFoundException if no such job could be found in the list
+     */
+    public boolean remove(Job toRemove) throws JobNotFoundException {
```
Who is throwing the Exception?

eugenepeh commented on src/main/java/seedu/address/model/job/Status.java:
```diff
@@ -0,0 +1,32 @@
+ * Represents the status of a car servicing job
+ */
+public class Status {
+    public static final String STATUS_ONGOING = "ongoing";
```
Is anything using this 2 constant?

eugenepeh commented on src/main/java/seedu/address/model/person/UniqueEmployeeList.java:
```diff
@@ -0,0 +1,140 @@
+        }
+
+        if (!target.equals(editedEmployee) && internalList.contains(editedEmployee)) {
+            throw new DuplicateEmployeeException();
```
Is it helpful to throw an exception that doesn't tells you anything?

eugenepeh commented on src/main/java/seedu/address/model/person/exceptions/DuplicateEmployeeException.java:
```diff
@@ -3,10 +3,10 @@
-    public DuplicatePersonException() {
+public class DuplicateEmployeeException extends DuplicateDataException {
+    public DuplicateEmployeeException() {
         super("Operation would result in duplicate persons");
```
Doesn't seems very useful to me unless you're showing the message to user, asking for confirmation.

eugenepeh commented on src/main/java/seedu/address/ui/MainWindow.java:
```diff
@@ -194,4 +202,40 @@ private void handleShowHelpEvent(ShowHelpRequestEvent event) {
+            themeName = themes[THEME_INDEX_TEAL];
+            break;
+        case 2:
+            themeName = themes[THEME_INDEX_DARK];
```
here seems abit weird. If you're using array constants... why need switch case?

eugenepeh reviewed (commented):
Just a brief review of your codes.

eugenepeh commented:
Reviewed in https://github.com/CS2103JAN2018-W09-B1/main/issues/74

yuhongherald committed:
edits

yuhongherald committed:
edit

yuhongherald committed:
edit

yuhongherald committed:
edit

yuhongherald committed:
edit

yuhongherald committed:
edit

yuhongherald committed:
edit

yuhongherald committed:
edit

yuhongherald committed:
edit

yuhongherald committed:
edit

yuhongherald committed:
edit

yuhongherald committed:
edits

yuhongherald committed:
edit

yuhongherald committed:
Rename method addUnreviewedJobEntries to addSheet

whenzei committed:
UserGuide update (#80)

whenzei committed:
UserGuide.adoc: minor update

charmaineleehc committed:
DeveloperGuide.adoc: update

yuhongherald committed:
Add importAll command (#78)

* fix merge errors

* Add dummy classes to support import feature

* Add details in JobEntry and SessionData

* more edits

* Add classes to support ImportSession

* Apply SLAP to make methods smaller

* Add ImportSession method to read excel file.

* Add SheetHeaderFields class

* Add SheetWithHeaderFields to extract row and column operations

* Finish ImportSession feature to write to file

* trailing whitespace

* Extract messages in ImportSession as constants. Write dummy ImportAllCommand, with functional support classes.

* Use unmodifiableList for getting lists in sessionData and jobEntry.

* Implement importAll command

* fix modelStub

* fix checkstyle error

* fix checkstyle error

* Rename method addUnreviewedJobEntries to addSheet

yuhongherald committed:
remove DG

yuhongherald committed:
Merge branch 'v1.3-importcommand'

yuhongherald committed:
Update DG

yuhongherald committed:
update DG

yuhongherald committed:
Merge pull request #15 from CS2103JAN2018-W09-B1/master

update

whenzei committed:
Add test for job adding feature (#85)

Richardson0694 committed:
Add archive command (#84)

* AddressBookParser.java: add in archive command
CliSyntax.java: add in PREFIX_DATE
ParserUtil.java: add in parseDate method
Model.java: add in archiveJob method
Date.java: add in constraints, regex test and new constructor
ArchiveCommand.java: created archive command
ArchiveCommandParser.java: created parser for archive command
DateRange.java: create DateRange object for archive command

* ModelManager.java: Add in archiveJob method

* Date.java: changed date format, included methods to getDay, getMonth, getYear
DateRange.java: add in compareTo method

* ArchiveCommand.java: change date prefix and remove extend to UndoableCommand
CommandWords.java: include in archive command
ArchiveCommandParser.java: change date prefix
CliSyntax.java: change date prefix and removed address prefix
AddressBook.java: implement archiveJob method
Date.java: change date format and date regex

* AddressBook.java: change job iterator

* MainApp.java: included archiveJobStorage and updated StorageManager constructor call
AddressBook.java: add getArchiveJobList method
ReadOnlyAddressBook.java: add abstract method of getArchiveJobList method
UserPrefs.java: included archiveJobFilePath
Storage.java: add abstract method of getArchiveJobPath, readArchiveJob and saveArchiveJob
StorageManager.java: updated StorageManager constructor to include ArchiveJobStorage, implemented getArchiveFilePath, readArchiveJob and saveArchiveJob
XmlFileStorage.java: add saveDataToFile and loadDataFromArchiveFile method
ArchiveJobStorage.java: interface to access archive data
XmlArchiveJobStorage.java: implement ArchiveJobStorage interface to access archive data
XmlSerializableArchiveJob.java: serialise archive job entries into xml file

* Updated test files

* FileUtil.java: add in createEvenIfExist method
AddressBook.java: create a new archiveJob list everytime the command is called
XmlArchiveJobStorage.java: replace createIfMissing method with createEvenIfExist method
Fix checkstyles

* Fix checkstyle

* Fix checkstyle

* Fix checkstyle

* Revert changes for Date.java
Fix checkstyle

* Changed format of Date
Minor edit

yuhongherald committed:
Add UML diagrams for set command and import/save

yuhongherald committed:
Add UML diagrams for set command and import/save (#88)

yuhongherald committed:
move session from model to storage

yuhongherald committed:
Add ImportSessionTest

yuhongherald committed:
modify test file

yuhongherald committed:
move session from model to storage (#91)

Add test case for ImportSession

charmaineleehc committed:
Add feature to connect to gmail and authorize (#83)

* gmail login

* update gradle

* remove import session test

yuhongherald committed:
Add documentation for import

yuhongherald committed:
Add design considerations

yuhongherald committed:
change sesionData get to return a job list instead of job entry list

whenzei committed:
DeveloperGuide.adoc: add implementation details for add job feature and theme setting feature

whenzei committed:
Add new UI panel to display details of selected job

whenzei committed:
Rename all commands and AddressBook (#96)

whenzei committed:
Update coveralls url

yuhongherald committed:
edit

yuhongherald committed:
resolve conflict

yuhongherald committed:
hard reset

yuhongherald committed:
Merge pull request #16 from CS2103JAN2018-W09-B1/master

update

yuhongherald committed:
Merge branch 'master' into v1.4-feature

# Conflicts:
#	src/main/java/seedu/carvicim/storage/session/SessionData.java

yuhongherald committed:
Merge branch 'master' into v1.4-documentation

yuhongherald committed:
Add aspect of supporting mass add methods addJobs/addMissingEmployees

yuhongherald committed:
resolve mrege conflict

whenzei committed:
Remove unused method and class (#97)

yuhongherald committed:
Process resources

yuhongherald committed:
Fix ImportSessionTest

yuhongherald committed:
checkstyle

yuhongherald committed:
Update UML diagrams

yuhongherald committed:
Add UML diagram for ImportSession closeSession()

yuhongherald committed:
edit

yuhongherald committed:
Add documentation for import (#92)

* fix merge errors

* Add dummy classes to support import feature

* Add details in JobEntry and SessionData

* more edits

* Add classes to support ImportSession

* Apply SLAP to make methods smaller

* Add ImportSession method to read excel file.

* Add SheetHeaderFields class

* Add SheetWithHeaderFields to extract row and column operations

* Finish ImportSession feature to write to file

* trailing whitespace

* Extract messages in ImportSession as constants. Write dummy ImportAllCommand, with functional support classes.

* Use unmodifiableList for getting lists in sessionData and jobEntry.

* Implement importAll command

* fix modelStub

* fix checkstyle error

* fix checkstyle error

* DG test

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edits

* edits

* edit

* edit

* edit

* edit

* edit

* edits

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edit

* edits

* edit

* Rename method addUnreviewedJobEntries to addSheet

* remove DG

* Update DG

* update DG

* Add UML diagrams for set command and import/save

* move session from model to storage

* Add ImportSessionTest

* modify test file

* Add documentation for import

* Add design considerations

* resolve conflict

* hard reset

* Add aspect of supporting mass add methods addJobs/addMissingEmployees

* Update UML diagrams

* Add UML diagram for ImportSession closeSession()

yuhongherald committed:
Merge branch 'v1.4-documentation'

yuhongherald committed:
Merge branch 'v1.4-documentation' into v1.4-feature

yuhongherald committed:
optimize code

yuhongherald committed:
fix ImportSession test case

yuhongherald committed:
edit test

yuhongherald committed:
Move methods, fields and constants from ImportSession to SessionData.

yuhongherald committed:
Add classes to support seesionData persistent data. WIP.

yuhongherald committed:
Merge branch 'master' into v1.4-feature

yuhongherald committed:
Revert "Add classes to support seesionData persistent data. WIP."

This reverts commit a9f0effb31b1e7be64b65bd2bb2f860787831bb7.

yuhongherald committed:
Merge branch 'v1.4-feature' of https://github.com/yuhongherald/addressbook-level4 into v1.4-feature

yuhongherald committed:
Add classes to support session data persistence. WIP.

yuhongherald committed:
Revert "Add classes to support session data persistence. WIP."

This reverts commit bebb8eaf0c3d4afd36f56f857257541debca4ff0.

yuhongherald committed:
Add import and save commands.

yuhongherald committed:
Add command to list jobs

yuhongherald committed:
Add acceptAll, rejectAll and switch commmands

yuhongherald committed:
Add accept and reject command.

yuhongherald committed:
uncomment check in date

yuhongherald committed:
Modify ModelStub

yuhongherald committed:
Remove unused imports

whenzei committed:
UserGuide.adoc: add details for job remark feature

whenzei committed:
Merge pull request #102 from whenzei/v1.4_documentation

UserGuide.adoc: add details for job remark feature

yuhongherald committed:
Add commands to support importing job entries individually (#99)

* Process resources

* Fix ImportSessionTest

* checkstyle

* Update UML diagrams

* Add UML diagram for ImportSession closeSession()

* edit

* optimize code

* fix ImportSession test case

* edit test

* Move methods, fields and constants from ImportSession to SessionData.

* Add classes to support seesionData persistent data. WIP.

* Revert "Add classes to support seesionData persistent data. WIP."

This reverts commit a9f0effb31b1e7be64b65bd2bb2f860787831bb7.

* Add classes to support session data persistence. WIP.

* Revert "Add classes to support session data persistence. WIP."

This reverts commit bebb8eaf0c3d4afd36f56f857257541debca4ff0.

* Add import and save commands.

* Add command to list jobs

* Add acceptAll, rejectAll and switch commmands

* Add accept and reject command.

* uncomment check in date

* Modify ModelStub

* Remove unused imports

yuhongherald committed:
Merge branch 'master' into v1.4-feature

yuhongherald committed:
Merge pull request #17 from CS2103JAN2018-W09-B1/master

update

yuhongherald committed:
Extract method to resetJobList

yuhongherald committed:
Update UserGuide.adoc to reflect new commands (#103)

* Add documentation for listj, switch, importAll

yuhongherald committed:
Remove unused imports

yuhongherald committed:
Extract resetJobList method (#104)

yuhongherald committed:
Fix bugs:
CommandWords toString does not enforce strict ordering.
SesseionData freeResources nullPointer.

Richardson0694 committed:
Documentation and updating for sort and archive command (#107)

whenzei committed:
Update coverage badge

charmaineleehc committed:
Send blank email (#101)

* email command

* update userguide

* collate

whenzei committed:
v1.4 update (#106)

* Rename commands

* Rename all AddressBook related classes and methods into Carvicim

* Remove unused method and class

* Fix checkstyle error

* Update delete employee feature to prevent deletion of employee that is assigned to a job

* Remove edit employee feature

* Update checkstyle error

* Add remark adding feature

* Fix checkstyle error

* Add test cases for delete employee feature

* Remove unused classes

* Add tests for XmlAdaptedJob

* Add missing method header in XmlAdaptedJobTest

* Remove test

* Fix sessionData nullpointer

* Fix CommandWords toString ordering problem

* Add missing imports

* Update methods name for DeleteEmployeeCommandTest

* Add back test for DeleteEmployeeCommandSystemTest

* Add new tests for RemarkCommand

* Add more tests for XmlAdaptedJob

* Update Clear command

* Add test for ListJobCommand

* Add tests for RemarkCommandParser

* Reaname SelectCommand to SelectEmployeeCommand

* Fix checkstyle error

* Add more tests for CarvicimParser

* Add close job feature

* UserGuide.adoc: update features

Richardson0694 committed:
Add analyse command (#110)

yuhongherald committed:
Merge branch 'master' into master

yuhongherald committed:
Merge pull request #18 from CS2103JAN2018-W09-B1/master

update

yuhongherald committed:
V1.4 bug fix (#112)

* remove undo support for some commands, fix import session check

* suport undo for commands

* fix bug

* remove unused imports

whenzei committed:
v1.4 code collation and some clean ups

whenzei committed:
Update UserGuide commands (#115)

whenzei committed:
Update model class diagram

whenzei committed:
Add more tests for CloseJobCommand and minor updates (#116)

whenzei committed:
Update README.adoc

whenzei committed:
Update README.adoc

whenzei committed:
Update UI image

whenzei committed:
Update UserGuide.adoc

charmaineleehc committed:
V1.4.2 (#117)

* user logged into carvicim gmail account

* emailcommand test case

* add some reference to figures

whenzei committed:
Fix typo

whenzei committed:
Disable remark feature (#119)

Richardson0694 committed:
Create test files for archive command (#118)

Richardson0694 committed:
Remove archive command status check (#120)

whenzei committed:
Fix remark feature (#124)

yuhongherald committed:
Merge pull request #19 from CS2103JAN2018-W09-B1/master

update

yuhongherald committed:
Import commands are fully undoable and persistent. (#151)

Other changes:
Remove save command.
Squashed a couple of bugs:
#131 #133 #146 #150

whenzei committed:
Bug fixes and V1.5 updates (#152)

* Update JobDisplayPanel UI
* Update UserPrefs to store running JobNumber
* Fix add employee allowing duplicates
* Update JobNumber to be consistent when undo an AddJobCommand
* Update CloseJob feature to change the status of a job instead of removing it
* Update JobList to filter for job with ongoing status
* Add more tests for XmlAdaptedRemark, CarvicimParser and ListJobCommand

whenzei committed:
Update formatting of UserGuide (#153)

whenzei committed:
UserGuide.adoc: update Quick Start section

yuhongherald committed:
gitignore test files

yuhongherald committed:
Merge branch 'master' into master

yuhongherald committed:
Merge pull request #20 from CS2103JAN2018-W09-B1/master

update

whenzei committed:
Add new FindJobCommand and ListOngoingJobCommand (#155)

yuhongherald committed:
Fix importSessionTest

yuhongherald committed:
Add ImportSessionTest

yuhongherald committed:
Apply SLAP

yuhongherald committed:
Apply SLAP

yuhongherald committed:
Update UG DG

yuhongherald committed:
Merge pull request #21 from CS2103JAN2018-W09-B1/master

update

yuhongherald committed:
Merge branch 'master' into v1.5rc-importcomments

yuhongherald committed:
fix test

yuhongherald committed:
Add comment argument to accept and reject commands.

yuhongherald committed:
Fix test error

yuhongherald committed:
Add acceptAllCommandParser, with comment support

yuhongherald committed:
Fix test

yuhongherald committed:
fix test

yuhongherald committed:
disable test

yuhongherald committed:
Add comments support for rejectAllCommand. Update equals for accept, reject, acceptAll and rejectAll.

yuhongherald committed:
fix test case

yuhongherald committed:
disable file write in test

yuhongherald committed:
Merge branch 'v1.5rc-importcomments' into v1.5-importcommandcomments

yuhongherald committed:
comment checkstyle

yuhongherald committed:
Merge branch 'v1.5rc-importcomments' into v1.5-importcommandcomments

yuhongherald committed:
Add import comments (#154)

* Fix importSessionTest

* Add ImportSessionTest

* Apply SLAP

* Apply SLAP

* Update UG DG

* fix test

* Fix test error

* Fix test

* fix test

* disable test

* fix test case

* disable file write in test

* comment checkstyle

yuhongherald committed:
Merge branch 'master' into master

yuhongherald committed:
Remove unused imports

yuhongherald committed:
Fix error

yuhongherald committed:
remove importAll command

yuhongherald committed:
remove errorneous testsheet

yuhongherald committed:
git ignore

yuhongherald committed:
remove test sheet

yuhongherald committed:
Add test sheet

yuhongherald committed:
Merge branch 'yuhongherald-v1.5-importcommentfix'

yuhongherald committed:
Revert "Add test sheet"

This reverts commit da154bd31b41996cc2e82c67e0b044b5df92a3fb.

yuhongherald committed:
Merge branch 'master' of https://github.com/yuhongherald/addressbook-level4

yuhongherald committed:
update

yuhongherald committed:
fix

yuhongherald committed:
fix

yuhongherald committed:
fix

yuhongherald committed:
fix

yuhongherald committed:
fix

yuhongherald committed:
fix

whenzei committed:
UserGuide.adoc: update formatting (#158)

yuhongherald committed:
remove excel sheet

yuhongherald committed:
Merge branch 'master' into v1.5fix

yuhongherald committed:
fix merge

yuhongherald committed:
commit

yuhongherald committed:
commit

yuhongherald committed:
change gradle file permission to 777

yuhongherald committed:
change filepath

yuhongherald committed:
Change travis to change permissions to 777

yuhongherald committed:
travis edit

yuhongherald committed:
indentation

yuhongherald committed:
disable importsessiontest check

yuhongherald committed:
change gradlew permissions path

yuhongherald committed:
revert

yuhongherald committed:
V1.5fix (#161)

comments supported, undo disabled

yuhongherald committed:
switch order of tests

yuhongherald committed:
credentials

yuhongherald committed:
remove credential

yuhongherald committed:
add credential

yuhongherald committed:
Merge branch 'master' into v1.5fix

yuhongherald committed:
import order

yuhongherald committed:
revert travis file

yuhongherald committed:
import session test fix

yuhongherald committed:
revert guisystemtests to systemtests

yuhongherald committed:
import order

yuhongherald committed:
revert gradle

yuhongherald committed:
revert build.gradle

yuhongherald committed:
fix test case

yuhongherald committed:
fix import command (#163)

* Fix importSessionTest

* Add ImportSessionTest

* Apply SLAP

* Apply SLAP

* Update UG DG

* fix test

* Add comment argument to accept and reject commands.

* Fix test error

* Add acceptAllCommandParser, with comment support

* Fix test

* fix test

* disable test

* Add comments support for rejectAllCommand. Update equals for accept, reject, acceptAll and rejectAll.

* fix test case

* disable file write in test

* comment checkstyle

* Remove unused imports

* Fix error

* remove importAll command

* fix

* fix

* fix

* fix

* fix

* fix

* remove excel sheet

* fix merge

* commit

* commit

* change gradle file permission to 777

* change filepath

* Change travis to change permissions to 777

* travis edit

* indentation

* disable importsessiontest check

* change gradlew permissions path

* revert

* switch order of tests

* credentials

* remove credential

* add credential

* import order

* revert travis file

* import session test fix

* revert guisystemtests to systemtests

* import order

* revert gradle

* revert build.gradle

* fix test case

whenzei committed:
Fix delete employee interaction with jobs (#164)

yuhongherald committed:
Merge branch 'v1.5fix'

# Conflicts:
#	CS2103-testsheet.xlsx
#	src/main/resources/.credentials/carvicim-gmail/StoredCredential

yuhongherald committed:
Merge pull request #22 from CS2103JAN2018-W09-B1/master

update

yuhongherald committed:
Add job list indicator panel

yuhongherald committed:
fix checkstyle errors

yuhongherald committed:
add job list indicator (#166)

whenzei committed:
Portfolio and DeveloperGuide update (#162)

whenzei committed:
Fix typo

yuhongherald committed:
Update portfolio and documentation (#167)

Richardson0694 committed:
Update and fix archive command (#165)

charmaineleehc committed:
find by tag feature (#168)

* find by tag feature

* findbytagcommandtest

charmaineleehc committed:
Email employees job details (#160)

* send email with job details

* add project portfolio page

Richardson0694 committed:
Delete ArchiveSequenceDiagram.png

Richardson0694 committed:
Fix and update analyse command (#170)

yuhongherald committed:
Merge pull request #23 from CS2103JAN2018-W09-B1/master

update

yuhongherald committed:
fix undo bug

yuhongherald committed:
[fix]garbage collect all the .files
[fix]Don't display commandwords(remove)
[fix]accept undo redo still doesn't work - use workbook/comments xxxx.temp (timestamp)
[fix]acceptAll and rejectAll still buggy (and put importAll back)

yuhongherald committed:
fix test error

yuhongherald committed:
add importAll command

yuhongherald committed:
add importSession test cases

yuhongherald committed:
fix checkstyle error

yuhongherald committed:
fix import session test

yuhongherald committed:
fix adoc

whenzei committed:
Add git ignore for email credentials

whenzei committed:
Remove StoredCredential

whenzei committed:
Update UI and add new Theme (#173)

whenzei committed:
Update non functional requirements

whenzei committed:
Fix Ui resolution problem (#174)

whenzei committed:
UI and Documentation updates (#175)

- Add new job selection feature
- Update UI
- DeveloperGuide.adoc: add implementation details
- Code collate

yuhongherald committed:
Merge pull request #24 from CS2103JAN2018-W09-B1/master

update

yuhongherald committed:
Merge branch 'master' of https://github.com/yuhongherald/addressbook-level4 into v1.5-fixbugs

# Conflicts:
#	src/main/resources/view/MainWindow.fxml
#	src/test/java/seedu/carvicim/logic/commands/AddJobCommandTest.java

yuhongherald committed:
update

yuhongherald committed:
V1.5 fixbugs (#172)

add importsession tests

fix multiple undo bug

whenzei committed:
Update UserGuide.adoc

whenzei committed:
UserGuide.adoc: update formatting and add screenshots (#179)

whenzei committed:
UserGuide.adoc: minor edits

whenzei committed:
Update portfolio (#180)

yuhongherald committed:
Merge branch 'master' into master

yuhongherald committed:
V1.5 code quality (#178)

* Add parser test cases

* add import session test cases

* extract common fields and methods

* add test cases for accept command

* Extract constants
Added logging

* Add switch and reject command test cases

* fix checkstyle

* Change helpWIndow to singleton design
Add Import test case

* fix checkstyle error

* Add test cases

* remove unused import

* accept all disable job number check

* minor modifications

* change error messages

* update collation

* update test case

* edit

* edit

* remove test case

* disable test cse

* edit

whenzei committed:
Update sequence diagram (#181)

Richardson0694 committed:
V1.5 (#182)

Richardson0694 committed:
Delete UiClassDiagram.png

Richardson0694 committed:
V1.5 update (#183)

whenzei committed:
V1.5 cleanup (#186)

charmaineleehc committed:
V1.5 (#176)

* more test cases for find by tag

* login command

* improve UGDG

* edited portfolio

* emailcommandtest

whenzei committed:
UserGuide.adoc: update email in command summary

whenzei committed:
DeveloperGuide.adoc: add more manual testing instructions (#188)

yuhongherald committed:
V1.5 documentation2 (#185)

* recollate code

* remove unused dependencies

* update readme to reflect apache poi

* Change set command message success

* update

* update javadocs

* edit

* update set command dg

* change

* edit

whenzei committed:
V1.5.2 documentation update (#191)

whenzei committed:
AboutUs.adoc: update role

Richardson0694 committed:
V1.5 documentation (#190)

Richardson0694 committed:
V1.5 minor edit (#192)

charmaineleehc committed:
V1.5.2 (#187)

* bug fix and add image to user guide

* UGDG improvements

Richardson0694 committed:
V1.5.1 (#197)

whenzei committed:
UserGuide.adoc: update command summary table

whenzei committed:
UserGuide.adoc: update UI orientation diagram (#199)

whenzei committed:
ReadMe.adoc: fix typo

whenzei committed:
README.adoc: fix typo in link

yuhongherald committed:
V1.5 documentation continued (#189)

* recollate code

* remove unused dependencies

* update readme to reflect apache poi

* Change set command message success

* update

* update javadocs

* edit

* update set command dg

* change

* edit

* edit

* update set command DG

* update set command dg

* update set command ug dg

* finish ug changes on import

* edit

* finish dg.
recollated codes

whenzei committed:
UserGuiide.adoc: clean up (#200)

charmaineleehc committed:
V1.5.3 (#198)

* coding standards fix

* update portfolio

yuhongherald committed:
update ppp (#201)

* recollate code

* remove unused dependencies

* update readme to reflect apache poi

* Change set command message success

* update

* update javadocs

* edit

* update set command dg

* change

* edit

* edit

* update set command DG

* update set command dg

* update set command ug dg

* finish ug changes on import

* edit

* finish dg.
recollated codes

* update portfolio

* update

* update

* edit

whenzei committed:
Clean up (#202)

yuhongherald committed:
update ppp

yuhongherald committed:
update ppp

Richardson0694 committed:
V1.5.1 (#204)

whenzei committed:
Update DeveloperGuide.adoc
